### DISCUSSION TITLE
WinUI 3 should be supported on Visual Studio 2019 16.9

### DISCUSSION THREAD

**Discussion by Alovchin91** [microsoft/WindowsAppSDK#568] [Ideas] [answered]
Hi,

According to the recently released Project Reunion 0.5 Preview's [documentation](https://docs.microsoft.com/en-us/windows/apps/project-reunion/#set-up-your-development-environment), it requires Visual Studio 2019 16.10 Preview to work with.

It might make sense for Project Reunion that is still in development, but it would be great if WinUI 3, which is going to be supported in Project Reunion 0.5 release later this month, was supported on a recently released Visual Studio 2019 16.9.

The reason for that is that Visual Studio 2019 16.9 is:

* The latest stable release of Visual Studio to date, which is unlikely to change before the WinUI 3 GA release;
* and Visual Studio 2019 16.9 is a [servicing baseline](https://docs.microsoft.com/en-us/visualstudio/releases/2019/servicing) which means many enterprises might decide to stick with it for the foreseeable future.

To include as many developers as possible, it would make sense to possibly backport some features required by WinUI 3 to VS 2019 16.9.x updates so that at least the go-live WinUI part of Project Reunion is fully supported by the time of release.

Thanks!

**Comment by BenJKuhn** [comment, marked as answer]
Thanks for reaching out. I got a bit more data about this since you asked me on Twitter. The specific changes relate to the developer experience. Visual Studio took tooling changes to enable WinUI in project Reunion to support features like XAML tree editing and "hot reloading" the XAML tree. These features won't work in 16.9. There are no other changes in VS 16.10 specifically to support Project Reunion. 

The Project Reunion package will install on Visual Studio 16.8 or higher, and other features should work on those versions. There are no plans at this time to backport support for the XAML debugging features to 16.9.

Ben

**Reply by Alovchin91** [reply]
Thanks for the swift response!
Would it then probably make sense to mention this in the documentation?

**Reply by anawishnoff** [reply]
I've put in a request to update the Project Reunion and WinUI 3 documentation to reflect this - thanks for calling it out.

**Reply by Alovchin91** [reply]
Yep, it's live 🙂 Thanks!
